Army/ArmyLRFull.cs

```diff
@@ -381,88 +324,72 @@ public void ArmyDiamondTokenOfDage(int quant)
         Core.RegisterQuests(4743);
         while (!Bot.ShouldExit && !Core.CheckInventory("Diamond Token of Dage", quant))
         {
-            Core.EquipClass(ClassType.Farm);
-            if (!Core.CheckInventory("Defeated Makai", 25))
-                Core.PrivateRoomNumber = 100000;
-                Core.KillMonster("tercessuinotlim", "m2", "Spawn", "Dark Makai", "Defeated Makai", 25, false, false,false);
-            Core.PrivateRoomNumber = Army.getRoomNr();
-            Core.EquipClass(ClassType.Solo);
+            Core.KillMonster("tercessuinotlim-999999", "m2", "Spawn", "Dark Makai", "Defeated Makai", 25, false, false, false);
             Adv.BestGear(GearBoost.Chaos);
-            ArmyHunt("aqlesson", new[] {"Carnax"}, "Carnax Eye", true, 1);
-            ArmyHunt("deepchaos", new[] {"Kathool"}, "Kathool Tentacle", true, 1);
-            ArmyHunt("dflesson", new[] {"Fluffy the Dracolich"}, "Fluffy's Bones", true, 1);
+            ArmyHunt("aqlesson", new[] { "Carnax" }, "Carnax Eye", ClassType.Solo, true, 1);
+            ArmyHunt("deepchaos", new[] { "Kathool" }, "Kathool Tentacle", ClassType.Solo, true, 1);
+            ArmyHunt("dflesson", new[] { "Fluffy the Dracolich" }, "Fluffy's Bones", ClassType.Solo, true, 1);
             Adv.BestGear(GearBoost.Dragonkin);
-            ArmyHunt("lair", new[] {"Red Dragon"}, "Red Dragon's Fang", true, 1);
+            ArmyHunt("lair", new[] { "Red Dragon" }, "Red Dragon's Fang", ClassType.Solo, true, 1);
             Adv.BestGear(GearBoost.Human);
-            ArmyHunt("bloodtitan", new[] {"Blood Titan"}, "Blood Titan's Blade", true, 1);
+            ArmyHunt("bloodtitan", new[] { "Blood Titan" }, "Blood Titan's Blade", ClassType.Solo, true, 1);
             Bot.Drops.Pickup("Legion Token", "Diamond Token of Dage");
```

Comment:
just do a Core.AddDrop(if it isnt in one already,) and a Bot.Wait.ForPickup("item"); at the end